### PR TITLE
Refactor Step isComplete

### PR DESCRIPTION
**Pull request by mcbouslog**
Package:
- lib-classifier

Describe your changes:
- add DrawingTask tests for isComplete view
- refactor Step.isComplete for incomplete drawing task (for below minimum marks or incomplete subtask)

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingTask.js [hunk 1]
```diff
@@ -48,8 +48,8 @@ export const Drawing = types.model('Drawing', {
       })
     },
 
-    get isComplete () {
-      return self.tools.reduce((isTaskComplete, tool) => isTaskComplete && tool.isComplete, true)
+    isComplete (annotation) {
+      return self.tools.reduce((isTaskComplete, tool) => isTaskComplete && tool.isComplete, true) && !!annotation?.isComplete
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingTask.js:52]: `annotation.isComplete` is always true for drawing annotations, so I think you don't need to check it here.

File: packages/lib-classifier/src/store/WorkflowStepStore/Step/Step.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ const baseStep = types
       let isIncomplete = false
       self.tasks.forEach(task => {
         const [annotation] = annotations.filter(annotation => annotation.task === task.taskKey)
-        isIncomplete = task.required && !annotation?.isComplete
+        isIncomplete = !task.isComplete(annotation)
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/WorkflowStepStore/Step/Step.js:29]: Should this be `isIncomplete && !task.isComplete(annotation)`? If the step has a few tasks, I think this is only returning the completeness of the last task, and ignoring the others.

Comment (mcbouslog) [reply packages/lib-classifier/src/store/WorkflowStepStore/Step/Step.js]: Yes! Great catch, thank you

File: packages/lib-classifier/src/store/WorkflowStepStore/Step/Step.spec.js [hunk 3]
```diff
@@ -59,6 +61,30 @@ describe('Model > Step', function () {
...
+      label: 'Point please.',
+      min: 1,
+      type: 'point'
+    }
+
+    before(function () {
+      tasks = [
+        MultipleChoiceTask.TaskModel.create(MultipleChoiceTaskFactory.build({ taskKey: 'T1', required: '' })),
+        SingleChoiceTask.TaskModel.create(SingleChoiceTaskFactory.build({ taskKey: 'T2', required: '' })),
+        DrawingTask.TaskModel.create(DrawingTaskFactory.build({ taskKey: 'T3', required: '', tools: [pointToolWithMin] }))
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/WorkflowStepStore/Step/Step.spec.js:78]: Small thing (not blocking), but `task.required` defaults to `false` in the project builder.

Comment (mcbouslog) [reply packages/lib-classifier/src/store/WorkflowStepStore/Step/Step.spec.js:78]: nice catch, thank you!

**Pull request merged**